types/parameterized_struct_type.go

```diff
@@ -0,0 +1,80 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/substrait-io/substrait-go/types/parameter_types"
+)
+
+// ParameterizedStructType is a struct having at least one parameter of type ParameterizedAbstractType
+// example: Struct<Decimal(P,S), INT8>.
+// If All arguments are concrete they are represented by StructType
+type ParameterizedStructType struct {
+	Nullability      Nullability
+	TypeVariationRef uint32
+	Type             []Type
+}
+
+func (*ParameterizedStructType) isRootRef() {}
+func (m *ParameterizedStructType) WithNullability(n Nullability) Type {
+	m.Nullability = n
+	return m
+}
```

Comment:
value receiver

Reply:
As mentioned above this is deliberate change to catch errors at compile time